### PR TITLE
Fix a bug in OffHeapStarTreeBuilder where file is not properly copied from data file to temp file

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Also add extra check to ensure that

### CODE REVIEW

File: pinot-core/src/main/java/com/linkedin/pinot/core/startree/OffHeapStarTreeBuilder.java [hunk 1]
```diff
@@ -645,9 +645,12 @@ private void constructStarTree(TreeNode node, int startDocId, int endDocId, int
...
       File tempFile = new File(_tempDir, startDocId + "_" + endDocId + ".unique.tmp");
-      try (FileChannel src = new FileInputStream(_dataFile).getChannel();
-          FileChannel dest = new FileOutputStream(tempFile).getChannel()) {
-        dest.transferFrom(src, startDocId * _docSize, tempBufferSize);
+      try (FileChannel src = new RandomAccessFile(_dataFile, "r").getChannel();
+          FileChannel dest = new RandomAccessFile(tempFile, "rw").getChannel()) {
+        long numBytesTransferred = src.transferTo(startDocId * _docSize, tempBufferSize, dest);
+        Preconditions.checkState(numBytesTransferred == tempBufferSize,
+            "Error transferring data from data file to temp file, transfer size mis-match");
+        dest.force(false);
```
Comment (kishoreg) [inline pinot-core/src/main/java/com/linkedin/pinot/core/startree/OffHeapStarTreeBuilder.java]: what is the idea behind calling force and especially with false?

Comment (Jackie-Jiang) [reply pinot-core/src/main/java/com/linkedin/pinot/core/startree/OffHeapStarTreeBuilder.java]: Ensure the data is flushed to disk (false means we don't enforce metadata to be flushed).

Comment (kishoreg) [reply pinot-core/src/main/java/com/linkedin/pinot/core/startree/OffHeapStarTreeBuilder.java]: this is in try catch block, which means close would be called at the end of it and that would internally call flush right?. btw, this if for my own understanding. I know having this additional line is probably harmless.

Comment (Jackie-Jiang) [reply pinot-core/src/main/java/com/linkedin/pinot/core/startree/OffHeapStarTreeBuilder.java]: @kishoreg Based on the javadoc, seems there is no guarantee on that.

**Pull request merged**